src/app/tests/TestEventPathParams.cpp

```diff
@@ -16,65 +16,120 @@
  *    limitations under the License.
  */
 
-/**
- *    @file
- *      This file implements unit tests for EventPathParams
- *
- */
-
 #include <app/EventPathParams.h>
-#include <lib/support/UnitTestRegistration.h>
-#include <nlunit-test.h>
+#include <app/util/mock/Constants.h>
+#include <lib/core/StringBuilderAdapters.h>
+#include <pw_unit_test/framework.h>
+
+using namespace chip::Test;
 
 namespace chip {
 namespace app {
 namespace TestEventPathParams {
-void TestSamePath(nlTestSuite * apSuite, void * apContext)
+
+class TestEventPathParams : public ::testing::Test
+{
+public:
+    static void SetUpTestSuite() { InitEventPaths(); }
+
+    static void InitEventPaths();
+
+    static chip::app::EventPathParams validEventpaths[6];
+};
+
+chip::app::EventPathParams TestEventPathParams::validEventpaths[6];
+
+TEST_F(TestEventPathParams, SamePath)
 {
     EventPathParams eventPathParams1(2, 3, 4);
     EventPathParams eventPathParams2(2, 3, 4);
-    NL_TEST_ASSERT(apSuite, eventPathParams1.IsSamePath(eventPathParams2));
+    EXPECT_TRUE(eventPathParams1.IsSamePath(eventPathParams2));
 }
 
-void TestDifferentEndpointId(nlTestSuite * apSuite, void * apContext)
+TEST_F(TestEventPathParams, DifferentEndpointId)
 {
     EventPathParams eventPathParams1(2, 3, 4);
     EventPathParams eventPathParams2(6, 3, 4);
-    NL_TEST_ASSERT(apSuite, !eventPathParams1.IsSamePath(eventPathParams2));
+    EXPECT_FALSE(eventPathParams1.IsSamePath(eventPathParams2));
 }
 
-void TestDifferentClusterId(nlTestSuite * apSuite, void * apContext)
+TEST_F(TestEventPathParams, DifferentClusterId)
 {
     EventPathParams eventPathParams1(2, 3, 4);
     EventPathParams eventPathParams2(2, 6, 4);
-    NL_TEST_ASSERT(apSuite, !eventPathParams1.IsSamePath(eventPathParams2));
+    EXPECT_FALSE(eventPathParams1.IsSamePath(eventPathParams2));
 }
 
-void TestDifferentEventId(nlTestSuite * apSuite, void * apContext)
+TEST_F(TestEventPathParams, DifferentEventId)
 {
     EventPathParams eventPathParams1(2, 3, 4);
     EventPathParams eventPathParams2(2, 3, 6);
-    NL_TEST_ASSERT(apSuite, !eventPathParams1.IsSamePath(eventPathParams2));
+    EXPECT_FALSE(eventPathParams1.IsSamePath(eventPathParams2));
 }
-} // namespace TestEventPathParams
-} // namespace app
-} // namespace chip
 
-namespace {
-const nlTest sTests[] = { NL_TEST_DEF("TestSamePath", chip::app::TestEventPathParams::TestSamePath),
-                          NL_TEST_DEF("TestDifferentEndpointId", chip::app::TestEventPathParams::TestDifferentEndpointId),
-                          NL_TEST_DEF("TestDifferentClusterId", chip::app::TestEventPathParams::TestDifferentClusterId),
-                          NL_TEST_DEF("TestDifferentEventId", chip::app::TestEventPathParams::TestDifferentEventId),
-                          NL_TEST_SENTINEL() };
+/*
+{kInvalidEndpointId, kInvalidClusterId, kInvalidEventId},
```

Comment:
I think that this comment should be removed

Reply:
I Think it is relevant though; makes what happened in InitEventPaths() more readable,
it shows that for example:
for validEventpaths[0] we have {kInvalidEndpointId, kInvalidClusterId, kInvalidEventId}, 
for validEventpaths[1] we have {kInvalidEndpointId, MockClusterId(1), kInvalidEventId},